proposal/package-importer.md

```diff
@@ -0,0 +1,191 @@
+
+## Summary
+
+> This section is non-normative.
+
+Sass users often need to use styles from a dependency to customize an existing
+theme or access styling utilities.
+
+This proposal defines a `pkg` URL scheme for usage with `@use` that directs an
+implementation to resolve a URL within a dependency. The implementation will
+resolve the dependency URL using the standard resolution for that environment.
+Once resolved, this URL will be loaded in the same way as any other `file:` URL.
+
+For example, `@use 'pkg:bootstrap';` would be able to resolve to the path of a
+library-defined export within the `bootstrap` dependency. In Node, that would be
+resolved within `node_modules`, using the [node resolution algorithm]. In Dart,
+that would be resolved within `pub-cache`, using [package-config].
+
+[node resolution algorithm]: https://nodejs.org/api/packages.html
+[package-config]: https://pub.dev/packages/package_config
+
+There will be tension between the different path resolution algorithms that are
+used. To address this, a `pkg:` url will only use the environment's resolution
+initially in order to resolve a starting path. After that, it will be handed off
+to the Sass algorithm, which works directly with the filesystem, to resolve
+index files and partials.
+
+To better understand and allow for testing against the recommended algorithm, a
+[Sass pkg: test] repository has been made with a rudimentary implementation of
+the algorithm.
+
+[Sass pkg: test]: https://github.com/oddbird/sass-pkg-test
+
+### Design Decisions
+
+We could use the `~` popularized by Webpack's `load-sass` format, but this has
+been deprecated since 2021. In addition, since this creates a URL that is
+syntactically a relative URL, it does not make it clear to the implementation or
+the reader where to find the file.
+
+While the Dart Sass implementation allows for the use of the `package:` url
+scheme, a similar standard doesn't exist in Node. We could add `node_modules` to
+the load path, but that again suffers from lack of clarity to the implementation
+and the reader.
+
+Some npm packages have adopted a convention of declaring their Sass entrypoints
+using `"style"` or `"sass"` keys in their `package.json`. This is tied to the
+npm context, and would require parsing package.json to find these files.
+Instead, each implementation should use the standard patterns and tooling
+(`require` for Node and `package_config` for Dart) for resolving file paths.
+This allows library authors control over what is public, which follows Sass's
+general design.
+
+This proposal does not add support for a [Node custom user condition]. Node
+currently does not provide a native way to set a condition on a single
+invocation of `require.resolve`, meaning we would need to implement our own
+`package.json` parsing similar to [Rollup] or add a dependency on a community
+solution like [resolve.exports]. However, this proposal will not prevent future
```

Comment:
I'm not intrinsically opposed to using something community-standard like `resolve.exports`, especially since it's so small.